### PR TITLE
RBAC commands changes for custom role and bug fixes

### PR DESCRIPTION
**Pull request by shuagarw**
-Handle cases for roles with duplicate names and multiple assignments for a particular case.
-Authorization package version update to use get role definition by name filter.
-Modify RoleDefinitionId to be displayed and taken Input just as a guid in role definition and assignment commands.
-Tests fixes and recordings.
-Added RoleId as a parameter in create and delete role assignment commands.

### CODE REVIEW

File: lib/commands/arm/providers/provider.operations._js [hunk 1]
```diff
@@ -29,23 +31,24 @@ exports.init = function (cli) {
   var providerOperations = provider.category('operations')
     .description($('Commands to get the operations or actions allowed by an Azure resource provider.'));
   
-  providerOperations.command('show [actionString]')
+  providerOperations.command('show [operationSearchString]')
     .description($('Show operations for the requested provider action string. The command takes as input an action string (with possible wildcard (*) character(s)) which determines the action details to display.'))
```
Comment (amarzavery) [inline lib/commands/arm/providers/provider.operations._js]: Show operations for the requested **operation search string**

File: lib/commands/arm/providers/provider.operations._js [hunk 2]
```diff
@@ -29,23 +31,24 @@ exports.init = function (cli) {
...
-    .usage('[options] <actionString>')
-    .option('-a --actionString <actionString>', $('The provider action string (with possible wildcard (*) characters). Example: "*" to get all actions for all providers, "Microsoft.Insights/*" to get actions for a particular provider, "*/virtualMachines/*" to get all actions that can be performed on virtual machines.'))
+    .usage('[options] <operationSearchString>')
+    .option('-o --operationSearchString <operationSearchString>', $('The provider action string (with possible wildcard (*) characters). Example: "*" to get all actions for all providers, "Microsoft.Insights/*" to get actions for a particular provider, "*/virtualMachines/*" to get all actions that can be performed on virtual machines.'))
```
Comment (amarzavery) [inline lib/commands/arm/providers/provider.operations._js]: replace action string with operation search string

File: lib/commands/arm/providers/provider.operations._js (hunk 1 above)
Comment (shuagarw) [inline lib/commands/arm/providers/provider.operations._js]: Resolved.

File: lib/commands/arm/providers/provider.operations._js (hunk 2 above)
Comment (shuagarw) [inline lib/commands/arm/providers/provider.operations._js]: Resolved.

**Pull request merged**